test/lib/shellcheck-services.nix

```diff
@@ -45,18 +45,17 @@ let
     # Return set of services ({ service1 = true; service2 = true; ... })
     # which are either defined or not defined within `sourcePrefix`, depending
     # on `shouldMatch`.
-    collectServices = shouldMatch: lib.listToAttrs (builtins.concatLists (zipListsWith (services: file:
+    collectServices = shouldMatch: lib.listToAttrs (builtins.concatLists (map (def:
       let
+        services = def.value;
+        inherit (def) file;
         isMatching = lib.hasPrefix sourcePrefix file;
       in
         # Nix has no boolean XOR, so use `if`
         lib.optionals (if shouldMatch then isMatching else !isMatching) (
           (map (service: { name = service; value = true; }) (builtins.attrNames services))
         )
-    # TODO-EXTERNAL:
-    # Use `systemdServices.definitionsWithLocations` when https://github.com/NixOS/nixpkgs/pull/189836
-    # is included in nixpkgs stable.
-    ) systemdServices.definitions systemdServices.files));
+    ) systemdServices.definitionsWithLocations));
```

Comment:
I usually have the output of the shellcheck service in my test logs, but somehow not with this PR. Does it still run correctly?

Reply:
Yes. To test this, add an error (e.g. line `a = b`) to a `preStart` script and run the test.
Or check the list of all checked scripts:
```bash
nix build --no-link --print-out-paths .#tests.default.driver.nodes.machine.config.test.shellcheckServices.runShellcheck
```